pkg/ccl/streamingccl/streamingest/alter_replication_job.go

```diff
@@ -198,6 +219,91 @@ func alterReplicationJobHook(
 		if err != nil {
 			return err
 		}
+
+		// If a source address is being provided, we're enabling replication into an
+		// existing virtual cluster. It must be inactive, and we'll verify that it
+		// was the cluster from which the one it will replicate was replicated, i.e.
+		// that we're reversing the direction of replication. We will then revert it
+		// to the time they diverged and pick up from there.
+		if alterTenantStmt.ReplicationSourceAddress != nil {
+			dstTenantID, err := roachpb.MakeTenantID(tenInfo.ID)
+			if err != nil {
+				return err
+			}
+
+			// Here, we try to prevent the user from making a few
+			// mistakes. Starting a replication stream into an
+			// existing tenant requires both that it is offline and
+			// that it is consistent as of the provided timestamp.
+			if tenInfo.ServiceMode != mtinfopb.ServiceModeNone {
+				return errors.Newf("cannot start replication for tenant %q (%s) in service mode %s; service mode must be %s",
+					tenInfo.Name,
+					dstTenantID,
+					tenInfo.ServiceMode,
+					mtinfopb.ServiceModeNone,
+				)
+			}
+
+			streamAddress := streamingccl.StreamAddress(srcAddr)
+			streamURL, err := streamAddress.URL()
+			if err != nil {
+				return errors.Wrap(err, "url")
+			}
+			streamAddress = streamingccl.StreamAddress(streamURL.String())
+
+			client, err := streamclient.NewStreamClient(ctx, streamingccl.StreamAddress(srcAddr), p.ExecCfg().InternalDB)
+			if err != nil {
+				return errors.Wrap(err, "creating client")
+			}
+			srcID, resumeTS, err := client.PriorReplicationDetails(ctx, roachpb.TenantName(srcTenant))
```

Comment:
at some point we'll probably need to add some version compatibility checks, but since we don't do that in the main create replication flow, we can punt this.